src/rules/indentRule.ts

```diff
@@ -121,6 +159,130 @@ function walk(ctx: Lint.WalkContext<Options>): void {
     }
 }
 
+function recursiveGetNodeAt(root: ts.Node, pos: number): ts.Node | undefined {
+    const result: ts.Node | undefined = root.getChildAt(pos);
+    if (result === undefined) {
+        ts.forEachChild(root, function cb(node: ts.Node): void {
+            ts.forEachChild(node, cb);
+        });
+    }
+    if (result !== undefined) {
+        return result;
+    }
+    return root;
+}
+
+function recursiveGetNodeAtEnd(root: ts.Node, pos: number): ts.Node | undefined {
+    if (root.getEnd() === pos) {
+        return root;
+    }
+    let result: ts.Node | undefined;
+    ts.forEachChild(root, function cb(node: ts.Node): void {
+        if (node.getFullStart() < pos && node.getEnd() >= pos) {
+            result = recursiveGetNodeAt(node, pos);
+        }
+        ts.forEachChild(node, cb);
+    });
+    return result;
+}
+
+function getNodeDeepth(node: ts.Node, line: string): number {
+    let result = 0;
+    let parent: ts.Node | undefined = node.parent;
+    const blockTypes: ts.SyntaxKind[] = [
+        ts.SyntaxKind.Block,
+        ts.SyntaxKind.Parameter,
+        ts.SyntaxKind.CaseBlock,
+        ts.SyntaxKind.EnumMember,
+        ts.SyntaxKind.CaseClause,
+        ts.SyntaxKind.JsxElement,
+        ts.SyntaxKind.ModuleBlock,
+        ts.SyntaxKind.TemplateSpan,
+        ts.SyntaxKind.DefaultClause,
+        ts.SyntaxKind.ClassDeclaration,
+        ts.SyntaxKind.ArrayLiteralExpression,
+        ts.SyntaxKind.ObjectLiteralExpression,
+    ];
+    if (isCloseElement(node, line)) {
+        result --;
+    }
+    if (isCloseParen(node, line)) {
+        result ++;
+    }
+    if (inStatementParen(node)) {
+        result ++;
+    }
+    while (parent !== undefined) {
+        if (blockTypes.indexOf(parent.kind) > -1) {
+            result++;
+        }
+        if (inStatementParen(parent)) {
+            result ++;
+        }
+        parent = parent.parent;
+    }
+    if (nodeAtOutside(node)) {
+        result--;
+    }
+    if (
+        node.parent !== undefined
+        && node.parent.kind === ts.SyntaxKind.CallExpression
+        && /^\s*\./.test(line)
```

Comment:
please add a comment explaining what this condition does. it took me a while to figure it out

Reply:
done